5.5/run-mysqld.sh

```diff
@@ -15,35 +23,24 @@ function usage {
 	exit 1
 }
 
-function valid_mysql_identifier {
-	local var="$1" ; shift
-	[[ "${var}" =~ $mysql_identifier_regex ]]
-}
-
-function valid_mysql_password {
-	local var="$1" ; shift
-	[[ "${var}" =~ $mysql_password_regex ]]
+function check_env_vars {
+	[[ "$MYSQL_USER"     =~ $mysql_identifier_regex ]] || usage "Invalid MySQL username"
+	[ ${#MYSQL_USER} -le 16 ] || usage "MySQL username too long (maximum 16 characters)"
+	[[ "$MYSQL_PASSWORD" =~ $mysql_password_regex   ]] || usage "Invalid password"
+	[[ "$MYSQL_DATABASE" =~ $mysql_identifier_regex ]] || usage "Invalid database name"
+	[ ${#MYSQL_DATABASE} -le 64 ] || usage "Database name too long (maximum 64 characters)"
+	if [ -v MYSQL_ROOT_PASSWORD ]; then
+		[[ "$MYSQL_ROOT_PASSWORD" =~ $mysql_password_regex ]] || usage "Invalid root password"
+	fi
 }
 
-valid_mysql_identifier "$MYSQL_USER"     || usage
-valid_mysql_password   "$MYSQL_PASSWORD" || usage
-valid_mysql_identifier "$MYSQL_DATABASE" || usage
-
 # Make sure env variables don't propagate to mysqld process.
-mysql_user="$MYSQL_USER" ; unset MYSQL_USER
-mysql_pass="$MYSQL_PASSWORD" ; unset MYSQL_PASSWORD
-mysql_db="$MYSQL_DATABASE" ; unset MYSQL_DATABASE
-
-# Root password.
-if [ "$MYSQL_ROOT_PASSWORD" ]; then
-	valid_mysql_password "$MYSQL_ROOT_PASSWORD" || usage
-	root_pass="$MYSQL_ROOT_PASSWORD"
-fi
-unset MYSQL_ROOT_PASSWORD
-
-# SCL in CentOS/RHEL 7 doesn't support --exec, we need to do it ourselves
-# The '|| exit 1' is here so -e doesn't propagate into scl_source.
-source scl_source enable mysql55 || exit 1
+function unset_env_vars {
```

Comment:
can you add comment here as well? explain why we need to unset these env vars.

also can't you just write a one-liner? does unset support multiple env vars?